### PR TITLE
[SYCL][L0] Report events as submitted, not running, until they are completed

### PR DESCRIPTION
**Pull request by smaslov-intel**

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -4962,8 +4962,12 @@ pi_result piEventGetInfo(pi_event Event, pi_event_info ParamName,
...
     // Level Zero has a much more explicit notion of command submission than
     // OpenCL. It doesn't happen unless the user submits a command list. We've
-    // done it just above so the status is at least PI_EVENT_RUNNING.
-    pi_int32 Result = PI_EVENT_RUNNING;
+    // done it just above so the status is at least PI_EVENT_SUBMITTED.
+    //
+    // NOTE: We currently cannot tell if command is currently running, so
+    // it will always show up "submitted" before it is finally "completed".
+    //
+    pi_int32 Result = PI_EVENT_SUBMITTED;
```
Comment (jandres742) [inline sycl/plugins/level_zero/pi_level_zero.cpp:4970]: would doing this mess with timings, for applications querying for the submit_time?

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:4970]: I don't think sp. This is returning "status", not time. So, an event status can now being either "submitted" or "completed". There is a separate interface for querying timestamps of a kernel for all "submit", "start", "end" times: https://github.com/intel/llvm/blob/836ceec36b578ea1374973528eb60a2ada7deaed/sycl/plugins/level_zero/pi_level_zero.cpp#L5104

**Comment by smaslov-intel** [comment]
> Failed Tests (1):
  SYCL :: Basic/memory-consumption.cpp

This is a know issue fix for which is et to come to xmain: https://github.com/intel/llvm/issues/9008

**Comment by smaslov-intel** [comment]
> Failed Tests (1):
SYCL :: Basic/memory-consumption.cpp

This is a know issue: https://github.com/intel/llvm/issues/9008

**Review by againull — APPROVED** [review]

**Pull request merged**